src/Core/hco_types_mod.F90

```diff
@@ -317,6 +317,7 @@ MODULE HCO_TYPES_MOD
      INTEGER                     :: Cat            ! Category
      INTEGER                     :: Hier           ! Hierarchy
      INTEGER                     :: ScalID         ! Scale factor ID
+     INTEGER                     :: srcGMaskID     ! global source mask ID
```

Comment:
@1Dandan: Use 
```F90
REAL(sp) :: srcGMaskID
```
in order to avoid repeated casting from integer to real on each (I,J,L) iteration.